### PR TITLE
[SPARK-29473][SQL] move statement logical plans to a new file

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

move the statement logical plans that were created for v2 commands to a new file `statements.scala`, under the same package of `v2Commands.scala`.

This PR also includes some minor cleanups:
1. remove `private[sql]` from `ParsedStatement` as it's in the private package.
2. remove unnecessary override of `output` and `children`.
3. add missing classdoc.

### Why are the changes needed?

Similar to https://github.com/apache/spark/pull/26111 , this is to better organize the logical plans of data source v2.

It's a bit weird to put the statements in the package `org.apache.spark.sql.catalyst.plans.logical.sql` as `sql` is not a good sub-package name in Spark SQL.

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @rdblue @dongjoon-hyun @brkyvz @imback82

**Comment by viirya** [comment]
retest this please

**Review by viirya — COMMENTED** [review]
looks reasonable.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+ * Instead of parsing directly to a v1 CreateTable that keeps metadata in CatalogTable, and then
+ * converting that v1 metadata to the v2 equivalent, the sql [[CreateTableStatement]] plan is
+ * produced by the parser and converted once into both implementations.
+ *
+ * Parsed logical plans are not resolved because they must be converted to concrete logical plans.
+ *
+ * Parsed logical plans are located in Catalyst so that as much SQL parsing logic as possible is be
+ * kept in a [[org.apache.spark.sql.catalyst.parser.AbstractSqlParser]].
+ */
+abstract class ParsedStatement extends LogicalPlan {
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala]: It's good but please mention explicitly about the removal `private[spark]`.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    ifExists: Boolean) extends ParsedStatement
+
+
+/**
+ * A DROP TABLE statement, as parsed from SQL.
+ */
+case class DropTableStatement(
+    tableName: Seq[String],
+    ifExists: Boolean,
+    purge: Boolean) extends ParsedStatement
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala]: It's a nice clean-up, but let's mention explicitly about the removal of the following in the PR description.
```scala
-
-  override def output: Seq[Attribute] = Seq.empty
-
-  override def children: Seq[LogicalPlan] = Seq.empty
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+/**
+ * A DROP VIEW statement, as parsed from SQL.
+ */
+case class DropViewStatement(
+    viewName: Seq[String],
+    ifExists: Boolean) extends ParsedStatement
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala]: ditto.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+/**
+ * A DESCRIBE TABLE tbl_name col_name statement, as parsed from SQL.
+ */
+case class DescribeColumnStatement(
+    tableName: Seq[String],
+    colNameParts: Seq[String],
+    isExtended: Boolean) extends ParsedStatement
+
+/**
+ * A DELETE FROM statement, as parsed from SQL.
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala]: Thank you for adding description.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (except a few comments on the PR description.) Pending Jenkins.
Thank you, @cloud-fan . It's a nice clean-up!

**Review by HyukjinKwon — APPROVED** [review]

**Comment by cloud-fan** [comment]
I've updated the PR description to mention the small cleanups. Thanks for catching them @dongjoon-hyun !

**Review by imback82 — APPROVED** [review]
LGTM. Thanks for the clean up.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**